hesiod/cfgparse/yamlparser.py

```diff
@@ -0,0 +1,118 @@
+from pathlib import Path
+from typing import Any, Dict, cast
+import yaml
+from copy import deepcopy
+
+from hesiod.cfgparse.cfgparser import ConfigParser
+
+CFGT = Dict[str, Any]
+
+
+class YAMLConfigParser(ConfigParser):
+    def load_cfg(self) -> CFGT:
```

Comment:
Could the code in `load_cfg` and other methods be general purpose for all parsers?